decode.go

```diff
@@ -649,12 +649,12 @@ func (d *decoder) scalar(n *Node, out reflect.Value) bool {
 				return true
 			}
 		case float64:
-			if !isDuration && resolved <= math.MaxInt64 && !out.OverflowInt(int64(resolved)) {
+			if !isDuration && resolved >= math.MinInt64 && resolved <= math.MaxInt64 && !out.OverflowInt(int64(resolved)) {
 				out.SetInt(int64(resolved))
```

Comment:
You are fixing float being lower than MinInt64 could you add a test for it ?

Thanks

Reply:
I think `int_underflow_float` case covers it.